app/commands/user/reputation_token/award_for_pull_request.rb

```diff
@@ -0,0 +1,77 @@
+require 'octokit'
+
+class User
+  class ReputationToken
+    class AwardForPullRequest
+      include Mandate
+
+      initialize_with :action, :github_username, :params
+
+      def call
+        award_reputation_to_author if merged?
+        award_reputation_to_reviewers if closed?
+      end
+
+      private
+      def award_reputation_to_author
+        user = User.find_by(github_username: github_username)
+
+        # TODO: decide what to do with user that cannot be found
+        Rails.logger.error "Missing author: #{github_username}" unless user
+        return unless user
+
+        User::ReputationToken::CodeContribution::Create.(user, external_link, repo, pr_id, reason)
+      end
+
+      def award_reputation_to_reviewers
+        reviews = octokit_client.pull_request_reviews(repo, pr_id)
+        reviewer_usernames = reviews.map { |reviewer| reviewer[:user][:login] }
+
+        reviewers = ::User.where(handle: reviewer_usernames)
+        reviewers.find_each do |reviewer|
+          User::ReputationToken::CodeReview::Create.(reviewer, external_link, repo, pr_id, 'reviewed_code')
+        end
+
+        # TODO: consider what to do with missing reviewers
+        missing_reviewers = reviewer_usernames - reviewers.map(&:handle)
+        Rails.logger.error "Missing reviewers: #{missing_reviewers.join(', ')}" if missing_reviewers.present?
+      end
+
+      def merged?
+        params[:merged].present? && params[:merged]
```

Comment:
Is this the Ruby way of doing this? I saw somewhere that some people use a `!!` to do something like this.

Reply:
`params[:merged].present?` is fine without the `&& ...`

`nil.present? == false`
`false.present? == false`
`true.present? == true`